prody/utilities/catchall.py

```diff
@@ -78,10 +87,18 @@ def calcModeClusters(similarities, n_clusters=0, linkage='all', method='tsne', c
         nclusts = range(2,10,1)
 
     if linkage != 'all':
-        if not isinstance(method, str):
-            raise TypeError('linkage must be an instance of str')
-        if method not in ['ward', 'average', 'complete', 'single']:
-            raise ValueError('linkage must be either \'ward\', \'average\', \'complete\', or \'single\'')
+        if isinstance(linkage, list):
+            if len(linkage) > 4:
+                raise ValueError('linkage must be one or more of: \'ward\', \'average\', \'complete\', or \'single\'')
+            for val in method:
+                if val.lower() not in ['ward', 'average', 'complete', 'single']:
+                    raise ValueError('linkage must be one or more of: \'ward\', \'average\', \'complete\', or \'single\'')
+                linkages = [ x.lower() for x in linkage ]
+        elif not isinstance(linkage, str):
+            raise TypeError('linkage must be an instance of str or list of strs')
+
+        if linkage not in ['ward', 'average', 'complete', 'single']:
```

Comment:
This will be tripped if you have a list there.

Reply:
This isn't resolved yet. You need to indent this line and the next one to make it be part of the str block.